x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -301,14 +301,28 @@ private static Map<String, RoleDescriptor> initializeReservedRoles() {
                     "Grants access to manage all index templates and all ingest pipeline configurations."
                 )
             ),
-            // reporting_user doesn't have any privileges in Elasticsearch, and Kibana authorizes privileges based on this role
             entry(
                 "reporting_user",
                 new RoleDescriptor(
                     "reporting_user",
                     null,
                     null,
-                    null,
+                    new RoleDescriptor.ApplicationResourcePrivileges[] {
+                        RoleDescriptor.ApplicationResourcePrivileges.builder()
+                            .application("kibana-.kibana")
+                            .resources("*")
+                            .privileges(
+                                "feature_discover.minimal_read",
+                                "feature_discover.generate_report",
+                                "feature_dashboard.minimal_read",
+                                "feature_dashboard.generate_report",
+                                "feature_dashboard.download_csv_report",
+                                "feature_canvas.minimal_read",
+                                "feature_canvas.generate_report",
+                                "feature_visualize.minimal_read",
+                                "feature_visualize.generate_report"
+                            )
+                            .build() },
                     null,
                     null,
                     MetadataUtils.getDeprecatedReservedMetadata("Please use Kibana feature privileges instead"),
```

Comment:
Since we're adding this I assume this is no longer considered a deprecated role, so I think this can be removed?

Reply:
Looks like the `description` might be outdated? Since we're doing this we might want to update it.

Reply:
> Since we're adding this I assume this is no longer considered a deprecated role, so I think this can be removed?

Good question!

@tsullivan Can correct me, but my understanding is that we still want to keep it deprecated, just to make sure it grants necessary application privileges. The preferred way should still be to [create a custom role](https://www.elastic.co/guide/en/kibana/current/secure-reporting.html#grant-user-access) with the least app privileges.

Reply:
BTW

> The preferred way should still be to [create a custom role](https://www.elastic.co/guide/en/kibana/current/secure-reporting.html#grant-user-access) with the least app privileges.

I think this is accurate, as granting the least app privileges should be recommended. Also worth mentioning, the documentation page linked here is in need of some updates for 8.x. It should offer more clarity about what the `xpack.reporting.roles.enabled` setting actually does.

Reply:
Pushed https://github.com/elastic/elasticsearch/pull/118058/commits/cbd07422b5d2abbfca1b439c8c576fafb6dbe2d4 and https://github.com/elastic/elasticsearch/pull/118058/commits/42170526c086cc4df0fd8ce5ad64e3ef4c7c9159, which remove deprecation warning and update docs and description of `reporting_user` role. Let me know if the updated description makes sense.